ClinGen/clincoded
# Connects to #1309. CaseControl viewer bug.

**Technical notes:**
1. The changes made in the `__init__.py` file are to address the issue in which the **otherGenes** and **otherPMIDs** objects were not embedded in the **caseControl** object.
2. The changes made in the `case_control_curation.js` file were mostly layout re-arrangement for the **Case Control view page only**. With the current view page layout in production, the respective CaseCohort panels and ControlCohort panels (e.g. "Demographics") are not always aligned to the top (or vertically) for visual comparison. See screenshots below for reference.
**[ Before ]**
![image](https://cloud.githubusercontent.com/assets/6956310/23630484/263c764e-0270-11e7-82dc-98ff347ae706.png)
**[ After ]**
![image](https://cloud.githubusercontent.com/assets/6956310/23630511/3d6801c6-0270-11e7-836d-b8d146e99362.png)
3. A bug has also been identified in the `case_control_curation.js` file. This bug is currently in production and it causes the **Control Cohort's** "Additional Information about Group" information not being captured in the database.
4. There are also CSS changes added to address display issues on the **Case Control view page only** in various screen widths. See screenshots below for reference.
**[ Before ]**
![image](https://cloud.githubusercontent.com/assets/1878194/23600758/ff687766-01fc-11e7-9384-c6f3b1e8fe39.png)
**[ After  ]**
![image](https://cloud.githubusercontent.com/assets/6956310/23631041/112bf746-0272-11e7-94e4-e7946f0b1256.png)

**Known issues:**
With the smaller screen width (no larger than 767px), the CaseCohort and ControlCohort panels are alternatingly stacked (**view page only**):
![image](https://cloud.githubusercontent.com/assets/6956310/23631348/ffaa2492-0272-11e7-8d8b-2ddadcc834ae.png)

**Steps to test:**
1. Create a new GDM and add a new CaseControl evidence.
2. On the CaseControl curation form page, fill out all the fields, particularly the ones in the Additional Information panels.
3. Save the curation form and then go to the CaseControl **view page**.
4. Expect to see the genes and PMIDs provided in the Additional Information panels being shown in the external linkouts.
5. Expect to see the data entered for "Additional Information about Group" for **Control Cohort** being displayed.
6. Expect to see the vertical alignments of panels in the CaseCohort and ControlCohort columns are consistent.
7. Go back to the CaseControl **edit page**. Change the screen/window width to less than 768px. Scroll down the form and expect to see the **h4** elements are now properly displayed and no longer overlapping with other text:
![image](https://cloud.githubusercontent.com/assets/6956310/23631563/c790a8be-0273-11e7-965a-5c594527099e.png)
